src/Workspaces/Core/Portable/Differencing/LongestCommonSubsequence.cs

```diff
@@ -100,6 +100,15 @@ protected double ComputeDistance(TSequence oldSequence, int oldLength, TSequence
                 return (oldLength == newLength) ? 0.0 : 1.0;
             }
 
+            // If the sequences differ significantly in size their distance will be very close to 1.0 
+            // (even if one is a strict subsequence of the other).
+            // Avoid running an expensive LCS algorithm on such sequences.
+            double lenghtRatio = (oldLength > newLength) ? oldLength / newLength : newLength / oldLength;
```

Comment:
"length".

Reply:
I have cherry-pick the change as is. I'll fix it in master but not here.